src/main/java/com/zaxxer/nuprocess/windows/WindowsProcess.java

```diff
@@ -437,23 +439,27 @@ boolean writeStdin(int transferred)
          }
       }
 
-      if (userWantsWrite.compareAndSet(true, false)) {
```

Comment:
`WindowsProcess` was already using the `compareAndSet` to clear the old request, but paired with a blind `userWantsWrite.set` of the `onStdinReady` result, which means `WindowsProcess` still exhibited the race.

I've flipped the handling here (`!userWantsWrite.compareAndSet`) to make this method and `BasePosixProcess.writeStdin(int, int)` flow more similarly. I'm happy to revert that if you prefer.